plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -414,6 +420,45 @@ public RemoteIterator<LocatedFileStatus> listLocatedStatus(Path path)
         return new S3ObjectsV2RemoteIterator(listPath(path, OptionalInt.empty(), ListingMode.SHALLOW_ALL));
     }
 
+    @Override
+    public void batchDeleteFiles(Collection<Path> paths)
+            throws IOException
+    {
+        try {
+            List<Path> pathsList = paths.stream().collect(toImmutableList());
+            int maxObjectsPerRequest = 1000;
+            List<Path> currentBatch = new ArrayList<>();
+            int currentIndex = 0;
+            for (int i = 0; i < pathsList.size(); i++) {
+                if (currentIndex < maxObjectsPerRequest) {
```

Comment:
`com.google.common.collect.Lists#partition` is your friend.

however, here you can use `Iterables.partition` and avoid conversion from Collection to a List

Reply:
lol :D